src/scenes/GrandpaScene.ts

```diff
@@ -0,0 +1,378 @@
+  private map: Phaser.Tilemaps.Tilemap
+  public character!: IntroductionCharacter
+  private wallsLayer: Phaser.Tilemaps.TilemapLayer
+  private furnituresLayer: Phaser.Tilemaps.TilemapLayer
+  private spawnPoints: Map<string, Phaser.Geom.Point> = new Map()
+  private doorImage: Phaser.GameObjects.Image
+  private doorCollider: Phaser.Physics.Arcade.Collider
+  private letterImage: Phaser.GameObjects.Image
+  private letterOpenImage: Phaser.GameObjects.Image
+  private letterOpenText: Phaser.GameObjects.Text
+  private mapImage: Phaser.GameObjects.Image
+  private pouchImage: Phaser.GameObjects.Image
+  private backpackImage: Phaser.GameObjects.Image
+  private itemsLeftToCollect: string[] = [Collectibles.Letter]
+
+  constructor() {
+    super({
+      key: "GrandpaScene",
+      physics: {
+        default: "matter",
+        arcade: {
+          gravity: { y: 0 },
+          // debug: true
+        },
+      },
+    })
+  }
+
+  preload() {
+    this.loadSkinSpriteSheet(Skin.DEFAULT)
+    this.loadSkinSpriteSheet(Skin.BLUE)
+    this.loadSkinSpriteSheet(Skin.KUMAMON)
+    this.loadSkinSpriteSheet(Skin.SUNGLASSES)
+
+    this.cursors = this.input.keyboard?.createCursorKeys()!
+  }
+
+  create() {
+    this.scene.run("game-ui")
+    this.scene.launch("settings-ui")
+
+    Singleton.getInstance().gameUi = this.scene.get("game-ui") as GameUI
+
+    this.map = this.make.tilemap({ key: "grandpa-home-map" })
+
+    createCharacterAnims(this.anims)
+
+    this.cameras.main.setBounds(-40, -50, 250, 400)
+
+    this.createLayers()
+    this.createAreas()
+    this.createObjects()
+    this.createInteractions()
+    this.spawnCharacter()
+    this.addColliders()
+    this.createEventListeners()
+  }
+
+  createLayers() {
+    const tileset0 = this.map.addTilesetImage("CuteRPG_Interior_custom", "tiles4")
+    const tileset1 = this.map.addTilesetImage("Interiors_free_16x16", "tiles5")
+
+    this.map.createLayer("Floor", [tileset0!, tileset1!])!
+    this.wallsLayer = this.map.createLayer("Walls", [tileset0!, tileset1!])!
+    this.furnituresLayer = this.map.createLayer("Furnitures", [tileset0!, tileset1!])!
+    this.map.createLayer("Objects", [tileset0!, tileset1!])!
+
+    this.map.getObjectLayer("SpawnPoints")!.objects.forEach((spawnPoint) => {
+      const x = spawnPoint.x
+      const y = spawnPoint.y
+      this.spawnPoints.set(spawnPoint.name, new Phaser.Geom.Point(x!, y!))
+    })
+  }
+
+  createAreas() {
+    Singleton.getInstance().areas = this.map.getObjectLayer("Areas")!.objects.map((area) => {
+      if (this.itemsLeftToCollect.includes(area.name.replace("tip", ""))) {
+        return new TipArea(area.x!, area.y!, area.width!, area.height!)
+      } else if (area.name == "areaDoor") {
+        return new GrandpaDoorArea(this, area.x!, area.y!, area.width!, area.height!)
+      } else {
+        return new Area(area.x!, area.y!, area.width!, area.height!)
+      }
+    })
+  }
+
+  createObjects() {
+    // Door
```

Comment:
No need for this comment, if you feel like separting the code create a function for this code

Reply:
Done 👍